src/dosext/mfs/mfs.c

```diff
@@ -1656,99 +1655,31 @@ calculate_drive_pointers(int dd)
 static int
 dos_fs_dev(struct vm86_regs *state)
 {
-  u_char drive_to_redirect;
   int redver;
 
   Debug0((dbg_fd, "emufs operation: 0x%08x\n", WORD(state->ebx)));
 
   if (WORD(state->ebx) == DOS_SUBHELPER_MFS_REDIR_INIT) {
     init_all_drives();
-    mach_fs_enabled = TRUE;
 
     lol = SEGOFF2LINEAR(state->es, WORD(state->edx));
-    sda = (sda_t) Addr(state, ds, esi);
+    sda = Addr(state, ds, esi);
     redver = state->ecx;
     Debug0((dbg_fd, "lol=%#x\n", lol));
-    Debug0((dbg_fd, "sda=%p\n", (void *) sda));
+    Debug0((dbg_fd, "sda=%p\n", sda));
+    Debug0((dbg_fd, "redver=%02d\n", redver));
 
     init_dos_offsets(redver);
 
-    SETWORD(&(state->eax), 1);
-
-    return (lol_cdsfarptr(lol).segment || lol_cdsfarptr(lol).offset) ? TRUE : FALSE;
-  }
-
-  if (WORD(state->ebx) == 0) {
```

Comment:
Hmm, but where do you re-introduce it, as
advertised? In another patch? Can this be
done in a single patch please?